server/routes/user.js

```diff
@@ -21,19 +22,57 @@ router.post('/register', (req, res) => {
         else {
           const newUser = new User({
             email: req.body.email,
-            password: req.body.password
+            password: req.body.password,
+            language: req.body.language
           });
           bcrypt.genSalt(12, (err, salt) => {
             bcrypt.hash(newUser.password, salt, (err, hash) => {
               if(err) throw err;
               newUser.password = hash;
-              newUser.save().then(user => res.json({ "success": `${user.email} successfully registered`}))
+              newUser.save().then(user => res.status(201).json({ success: `${user.email} successfully registered`}))
                             .catch(err => console.error(err));
             });
           });
         }
       });
 });
 
+router.post('/login', (req, res) => {
+  const {validationErrors, isLoginValid} = loginValidator(req.body);
+
+  if(!isLoginValid) {
+    return res.status(400).json(validationErrors);
+  }
+
+  User.findOne({ email: req.body.email })
+      .then(user => {
+        if(!user) {
+          return res.status(404).json({ missingData: "User not found"});
+        }
+        bcrypt.compare(req.body.password, user.password)
+              .then(isPasswordValid => {
+                if(isPasswordValid) {
+                  const payload = {
+                    id: user._id,
+                    email: user.email
+                  };
+                  jwt.sign(payload, secretKey["secretOrKey"], {
+                    expiresIn: 86400 //expires in 1 day
+                  },
+                  (err, token) => {
```

Comment:
unclear where token is coming from (definition). should a var be created for the result of jwt.sign() and if so, should the `,` in line 61 be `;`?

Reply:
if you can verify that the token is getting generated and returned on the endpoint, then I think it is good to go

Reply:
Here is the result of successful login:
```
{
    "success": "hello22@yahoo.com is logged in successfully",
    "token": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlYzZjMTA1ZTA2ZTYxMjYwMDUyN2Y0NyIsImVtYWlsIjoiaGVsbG8yMkB5YWhvby5jb20iLCJpYXQiOjE1OTAwOTgyODUsImV4cCI6MTU5MDE4NDY4NX0.Nt-BA5JfaGiCsSVdQu_61aniFUyyKk3FKBdZIGDb9_g"
}
```

Reply:
> unclear where token is coming from (definition). should a var be created for the result of jwt.sign() and if so, should the `,` in line 61 be `;`?

This is an async way of generating token. Take a look at jsonwebtoken npm doc:
https://www.npmjs.com/package/jsonwebtoken
Jump to the section called 'Sign asynchronously'